front/lib/api/workspace.ts

```diff
@@ -7,15 +7,30 @@ import {
   Workspace,
 } from "@app/lib/models";
 import { MembershipInvitationType } from "@app/types/membership_invitation";
-import { UserType } from "@app/types/user";
+import { UserType, WorkspaceType } from "@app/types/user";
 
-export async function isWorkspaceAllowedOnDomain(wId: string) {
+export async function getWorkspaceInfos(
```

Comment:
Not sure we're okay with exposing a way to get a WorkspaceType outside of auth. 
I only the workspace name and allowed domain, and thought this was better than introducing a new type but can be challenged!